in-progress/8404-based-fallback.md

```diff
@@ -0,0 +1,191 @@
+
+Another approach might be that we allow a committee to exit the fallback mode if they provide an epoch proof (does not need to include blocks for every slot).
+
+However, this essentially make the committee a participant in the race to propose a block in the based fallback, and as they have the overhead of consensus they would be at a disadvantage.
+
+Furthermore, if they need to build an epoch proof without there being blocks on L1 to point against, we would need to support an extra flow in the node for them.
+
+A separate direction that I would like to entertain, is that we have two ways to "get out" of the based fallback, both based on time, similarly to how we ended up in here.
+
+1. If the last proven block is older than $T_{\textsf{fallback}, \textsf{exit}, \textsf{activity}}$ we allow exiting the based fallback.
+2. After $T_{\textsf{fallback}, \textsf{exit}}$ we allow exiting the fallback regardless of when the last proven block were.
+
+Option 1, ensures that we can quickly leave the based fallback if there is no activity, allowing a good experience for the happy path.
+Option 2, ensures that we will not be stuck forever in based fallback even if there is a malicious entity pushing blocks once in a while.
+As a bonus, they are simple rules so my 🥜🧠 can deal with them.
+
+![alt text](images/8404/image_4.png)
+
+### Relation to forced inclusion
+
+As mentioned, one of the cases where we enter the based fallback is if the validator set is massively censoring.
+If the L1 contract enforce that there is a minimum block size, the committees only option to try and censor will be to entirely stall the rollup.
+
+After $T_{\textsf{fallback}, \textsf{enter}}$ time that would push us into the based fallback.
+At this point, anyone can push the blocks with the censored transactions.
+Even if the committee is trying to limit the throughput by only pushing the minimum blocks, $T_{\textsf{fallback}, \textsf{exit}}$ can then be picked such that a minimum throughput could be supported over the duration.
+The minimum throughput here under the assumption that either they will need to push minimum blocks as fast as they can, or someone else might push a bigger block exiting even more.
+
+### Changes to existing architecture
+
+The changes should mostly be on the L1 contract and the sequencer client, with a minor potential change for the rollup circuit (might simply not be necessary).
+
+- The L1 contract need to deal with the flows as above.
+- The sequencer client need to know when fallback is activated such that it does not try to act as it normally would.
+  - I believe simply doing nothing is best here, as the fallbacks can be seen as a fully separate type of sequencer then.
+- Add a new "fallback" sequencer, which skips a lot of the checks of the current sequencer but mainly build and prove blocks.
+- Tee L1 contract/rollup circuit mix needs to support individual block proofs for this specific mode.
```

Comment:
Could they just stop at the BlockRoot rollup? Or do an epoch of size 1?

Reply:
Yes, the simplest would be just stopping at the block root and then publish that one 👍